src/parsec_client/operations/psa_algorithm.md

```diff
@@ -0,0 +1,347 @@
+# PSA Algorithm
+
+The algorithm is used to select the specific cryptographic operation and to set a policy on a new
+key.
+
+*Notice:* not all possible attributes are currently supported by Parsec. Please see the [API
+coverage](../../api_coverage.md) for an overview of what Parsec currently supports.
+
+## Algorithm message
+
+| Field name | Type    | Description                                 |
+|------------|---------|---------------------------------------------|
+| `variant`  | `oneof` | Key type, see below for the possible values |
+
+| `variant` oneof field name | Type                   |
+|----------------------------|------------------------|
+| `none`                     | `None`                 |
+| `hash`                     | `Hash`                 |
+| `mac`                      | `Mac`                  |
+| `cipher`                   | `Cipher`               |
+| `aead`                     | `Aead`                 |
+| `asymmetric_signature`     | `AsymmetricSignature`  |
+| `asymmetric_encryption`    | `AsymmetricEncryption` |
+| `key_agreement`            | `KeyAgreement`         |
+| `key_derivation`           | `KeyDerivation`        |
+
+### Nested Types
+
+#### None message
+
+An invalid algorithm identifier value. None does not allow any cryptographic operation with the key.
+The key can still be used for non-cryptographic actions such as exporting, if permitted by the usage
+flags.
+
+#### Hash enumeration
+
+Possible hash algorithms.
+
+- `HASH_NONE`: this default variant should not be used.
+- `MD2`: MD2. Deprecated: the MD2 hash is weak and deprecated and is only recommended for use in
+   legacy protocols.
```

Comment:
Do you think it's worth mentioning that it will be at the discretion of the system admin whether legacy algorithms are permitted?

Reply:
Yes good point! Will add it in the overview page.